src/env-inl.h

```diff
@@ -325,6 +326,44 @@ inline void Environment::set_printed_error(bool value) {
   printed_error_ = value;
 }
 
+inline void Environment::print_sync_warn() const {
+  if (!warn_on_sync_)
+    return;
+
+  v8::HandleScope handle_scope(isolate_);
+  v8::Local<v8::StackTrace> stack =
+      v8::StackTrace::CurrentStackTrace(isolate_,
+                                        10,
+                                        v8::StackTrace::kDetailed);
+
+  fprintf(stderr, "WARNING: Detected use of sync API\n");
+  for (int i = 0; i < stack->GetFrameCount() - 1; i++) {
+    v8::Local<v8::StackFrame> sf = stack->GetFrame(i);
+    node::Utf8Value fn_name_s(isolate_, sf->GetFunctionName());
+    node::Utf8Value sn_s(isolate_, sf->GetScriptName());
+    int ln = sf->GetLineNumber();
+    int cn = sf->GetColumn();
+
+    if (sf->IsEval()) {
+      if (sf->GetScriptId() == v8::Message::kNoScriptIdInfo)
+        fprintf(stderr, "    at [eval]:%i:%i\n", ln, cn);
+      else
+        fprintf(stderr, "    at [eval] (%s:%i:%i)\n", *sn_s, ln, cn);
+      return;
+    }
+
+    if (strcmp(*fn_name_s, "") == 0) {
+      fprintf(stderr, "    at %s:%i:%i\n", *sn_s, ln, cn);
+    } else {
+      fprintf(stderr, "    at %s (%s:%i:%i)\n", *fn_name_s, *sn_s, ln, cn);
+    }
+  }
+}
```

Comment:
This method is kind of stretching the limit of what belongs in a header file.  Maybe it's time to introduce src/env.cc.